### ISSUE TITLE
About message filter, that's gone too far from the community game

### ISSUE THREAD

**Issue by Nyte-BK201** [ddnet/ddnet#3027]
It was a bad mood entering the game seeing filtered messages. Those words are not malicious or harmful. The game just has gone too far from the community game.
Wasn't a good signal, the filter arises the feeling that leaving this game. The inertia still holds, maybe next month I am gone and store the memory without ***worlds. I don't see any necessity, but I know once the feature is added, it won't rewind back.

It's just a complaint, I am not arguing anymore.

**Comment by Nyte-BK201** [comment]
I know somebody is working on the filter but it is sad to see /practice still in the same place while adding the unpleasant feature.

**Comment by def-** [comment]
I'm assuming this is about the censoring filter we have enabled on DDNet CHN. It is supposed to be required by Chinese government in order to prevent certain terms being used. This will not be used on any other servers than CHN. What words in the filter do you have a problem with?

**Comment by Nyte-BK201** [comment]
> I'm assuming this is about the censoring filter we have enabled on DDNet CHN. It is supposed to be required by Chinese government in order to prevent certain terms being used. This will not be used on any other servers than CHN. What words in the filter do you have a problem with?

That is good feedback. I send messages to ask ACTom about the details but I have not heard any messages of "required by CHN government". If it is true, I would apologize here regarding the lack of a survey. The censoring filter, however, covers most political, f-words, and adult related stuff. The truth is, I do not meet so many censoring in dota2 CHN which feels like chatting with face masks.

**Comment by def-** [comment]
Yeah, I feel like insults should also not be part of the filter. We should only remove what is required by government. The problem is that the DDNet CHN servers are run by volunteers in China, so if we go against the law they will get into trouble, not me.  @ACTom thoughts?

**Comment by ACTom** [comment]
因为提问者是国人的原因，所以我就用中文回复了。
关键词过滤是我以及目前国服管理者的想法，基于以下几点：
1、随着玩家的增多，国外对华不友好的玩家也出现了，部分言论对于国内玩家非常不友好；
2、我们管理都非全职，没有办法随时去处理相关言论；
3、政治敏感和色情一直是国内互联网的红线，你可以咨询任何一个游戏公司以及云主机厂商；
4、tws的协议完全没有加密，意味着聊天内容是明文传输的，敏感词的传输是可以轻易被抓捕的；
基于以上原因以及对服主的保护，同时尽可能的减少对游戏的干扰，所以加入了相关功能。

**Comment by heinrich5991** [comment]
Google translate:
> Because the questioner is a Chinese, so I replied in Chinese.
> Keyword filtering is the idea of me and the current national service managers, based on the following points:
> 1. With the increase of players, foreign players who are not friendly to China have also appeared, and some comments are very unfriendly to domestic players;
> 2. Our management is not full-time, and there is no way to deal with relevant speech at any time;
> 3. Political sensitivity and pornography have always been the red lines of the domestic Internet. You can consult any game company and cloud host manufacturer;
> 4. The tws protocol is not encrypted at all, which means that the chat content is transmitted in plain text, and the transmission of sensitive words can be easily captured;
Based on the above reasons and the protection of the host, while reducing the interference to the game as much as possible, related functions have been added.

@ACTom Can you remove words from the filter which are merely 'very unfriendly to domestic players'? It's not a requirement from the Chinese government that people be nice to each other, is it?

**Comment by heinrich5991** [comment]
Also, what about pornography?

I doubt "fuck" etc. are used on the English servers in some sort of sexual meaning. Can you elaborate why these are blocked on Chinese servers, was there actual 'pornographic' or 'sexual' stuff happening or just words being used?

**Comment by ACTom** [comment]
> 
> 
> Google translate:
> 
> > Because the questioner is a Chinese, so I replied in Chinese.
> > Keyword filtering is the idea of me and the current national service managers, based on the following points:
> > 
> > 1. With the increase of players, foreign players who are not friendly to China have also appeared, and some comments are very unfriendly to domestic players;
> > 2. Our management is not full-time, and there is no way to deal with relevant speech at any time;
> > 3. Political sensitivity and pornography have always been the red lines of the domestic Internet. You can consult any game company and cloud host manufacturer;
> > 4. The tws protocol is not encrypted at all, which means that the chat content is transmitted in plain text, and the transmission of sensitive words can be easily captured;
> >    Based on the above reasons and the protection of the host, while reducing the interference to the game as much as possible, related functions have been added.
> 
> @ACTom Can you remove words from the filter which are merely 'very unfriendly to domestic players'? It's not a requirement from the Chinese government that people be nice to each other, is it?

Keyword filtering should be said to be an unwritten rule in China. The vocabulary we are currently using is obtained from a game company that is operating in China, and they are based on the vocabulary passed to them by the government. It’s composed of some words that their games banned.
The person in charge of this game company was once called by the government to talk for failing to delete pornographic comments on game forums in time.
Due to the above reasons, the scope of this vocabulary is too wide, so Dan_cao and I have manually deleted many words, and will also delete more inappropriate words based on player feedback.

> Also, what about pornography?
Pornographic vocabulary is strictly limited in China, including games and novels are forbidden to use, but this degree is really not easy to grasp, because there is no clear official thesaurus, and the government has launched many times for pornographic reasons. The "Net Net Action" aims to provide young people with a healthy online environment.

> I doubt "fuck" etc. are used on the English servers in some sort of sexual meaning. Can you elaborate why these are blocked on Chinese servers, was there actual 'pornographic' or 'sexual' stuff happening or just words being used?
Regarding the word "fuck", it is more of an insulting vocabulary in China. For insulting vocabulary, we currently have no unified opinion within us, so we have not deleted it from the list for the time being.

**Comment by Robyt3** [comment]
Discussion has concluded and this issue doesn't seem actionable to me.
